### PR TITLE
Add clean-state tests

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

This PR adds a circleCI step that packs every package, tries to install it in an empty TS repo and checks that no compilation errors occur. Most of the packages needed web3/ethers types and a lot of beginners tended to forget about that we decided to move to a centralized typings repo (like DefinitelyTypes) for those types. Now you only need to add it to typeRoots. It's already a dependency (not dev-dependency) of each package that needs it. 
```json
"compilerOptions": {
    "typeRoots": ["node_modules/@0xproject/typescript-typings/types", "node_modules/@types"],
}
```

## Motivation and Context

This will make sure, that the packages are not broken before we publish

## How Has This Been Tested?

On CI

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [x] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [] Added new entries to the relevant CHANGELOGs.
*   [ ] Updated the new versions of the changed packages in the relevant CHANGELOGs.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [x] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/monorepo-scripts/src/test_installation.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        fs.writeFileSync(indexFilePath, `import * as Package from '${packageName}'`);
+        const tsConfig = {
+            compilerOptions: {
+                module: 'commonjs',
+                target: 'es5',
+                lib: ['es2017', 'dom'],
+                declaration: true,
+                noImplicitReturns: true,
+                pretty: true,
+                strict: true,
```
Comment (fabioberger) [inline packages/monorepo-scripts/src/test_installation.ts]: Isn't the `typeRoot` param needed?

File: .circleci/config.yml [hunk 2]
```diff
@@ -27,6 +27,16 @@ jobs:
           key: repo-{{ .Environment.CIRCLE_SHA1 }}
           paths:
             - ~/repo
+  test-installation:
+    docker:
+      - image: circleci/node:6.12
```
[the remark this answers is not in the document]
Comment (LogvinovLeon) [reply .circleci/config.yml]: That's the version that all the other tests use. I've just copied that

File: README.md [hunk 3]
```diff
@@ -18,35 +18,27 @@ This repository is a monorepo including the 0x protocol smart contracts and nume
...
+| [`@0xproject/types`](/packages/types)                           | [![npm](https://img.shields.io/npm/v/@0xproject/types.svg)](https://www.npmjs.com/package/@0xproject/types)                           | Shared type declarations                                                    |
+| [`@0xproject/typescript-typings`](/packages/typescript-typings) | [![npm](https://img.shields.io/npm/v/@0xproject/typescript-typings.svg)](https://www.npmjs.com/package/@0xproject/typescript-typings) | Repository of types for external packages                                   |
+| [`@0xproject/utils`](/packages/utils)                           | [![npm](https://img.shields.io/npm/v/@0xproject/utils.svg)](https://www.npmjs.com/package/@0xproject/utils)                           | Shared utilities                                                            |
+| [`@0xproject/web3-wrapper`](/packages/web3-wrapper)             | [![npm](https://img.shields.io/npm/v/@0xproject/web3-wrapper.svg)](https://www.npmjs.com/package/@0xproject/web3-wrapper)             | Web3 wrapper                                                                |
```
[the remark this answers is not in the document]
Comment (LogvinovLeon) [reply README.md]: Spacing

File: package.json [hunk 4]
```diff
@@ -8,7 +8,9 @@
         "prettier": "prettier --write '**/*.{ts,tsx,json,md}' --config .prettierrc",
         "prettier:ci": "prettier --list-different '**/*.{ts,tsx,json,md}' --config .prettierrc",
         "report_coverage": "lcov-result-merger 'packages/*/coverage/lcov.info' | coveralls",
+        "test:installation": "node ./packages/monorepo-scripts/lib/test_installation.js",
```
[the remark this answers is not in the document]
Comment (LogvinovLeon) [reply package.json]: It's written in TS and compiled down to JS

File: .circleci/config.yml [hunk 5]
```diff
@@ -27,6 +27,16 @@ jobs:
...
+    working_directory: ~/repo
+    steps:
+      - restore_cache:
+          keys:
+            - repo-{{ .Environment.CIRCLE_SHA1 }}
+      - run: yarn lerna:exec 'yarn pack --filename package.tgz'
```
Comment (fabioberger) [inline .circleci/config.yml]: @LogvinovLeon

**Pull request merged**